// upstox-js-sdk/dist/model/AnalyticsData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AnalyticsData = void 0;
var _ApiClient = require("../ApiClient");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The AnalyticsData model module.
 * @module model/AnalyticsData
 * @version v0
 */
class AnalyticsData {
  /**
   * Constructs a new <code>AnalyticsData</code>.
   * @alias module:model/AnalyticsData
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>AnalyticsData</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/AnalyticsData} obj Optional instance to populate.
   * @return {module:model/AnalyticsData} The populated <code>AnalyticsData</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new AnalyticsData();
      if (data.hasOwnProperty('vega')) obj.vega = _ApiClient.ApiClient.convertToType(data['vega'], 'Number');
      if (data.hasOwnProperty('theta')) obj.theta = _ApiClient.ApiClient.convertToType(data['theta'], 'Number');
      if (data.hasOwnProperty('gamma')) obj.gamma = _ApiClient.ApiClient.convertToType(data['gamma'], 'Number');
      if (data.hasOwnProperty('delta')) obj.delta = _ApiClient.ApiClient.convertToType(data['delta'], 'Number');
      if (data.hasOwnProperty('iv')) obj.iv = _ApiClient.ApiClient.convertToType(data['iv'], 'Number');
      if (data.hasOwnProperty('pop')) obj.pop = _ApiClient.ApiClient.convertToType(data['pop'], 'Number');
    }
    return obj;
  }
}

/**
 * @member {Number} vega
 */
exports.AnalyticsData = AnalyticsData;
AnalyticsData.prototype.vega = undefined;

/**
 * @member {Number} theta
 */
AnalyticsData.prototype.theta = undefined;

/**
 * @member {Number} gamma
 */
AnalyticsData.prototype.gamma = undefined;

/**
 * @member {Number} delta
 */
AnalyticsData.prototype.delta = undefined;

/**
 * @member {Number} iv
 */
AnalyticsData.prototype.iv = undefined;

/**
 * @member {Number} pop
 */
AnalyticsData.prototype.pop = undefined;
