// upstox-js-sdk/dist/model/GetOptionContractResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GetOptionContractResponse = void 0;
var _ApiClient = require("../ApiClient");
var _InstrumentData = require("./InstrumentData");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The GetOptionContractResponse model module.
 * @module model/GetOptionContractResponse
 * @version v0
 */
class GetOptionContractResponse {
  /**
   * Constructs a new <code>GetOptionContractResponse</code>.
   * @alias module:model/GetOptionContractResponse
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>GetOptionContractResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/GetOptionContractResponse} obj Optional instance to populate.
   * @return {module:model/GetOptionContractResponse} The populated <code>GetOptionContractResponse</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new GetOptionContractResponse();
      if (data.hasOwnProperty('status')) obj.status = _ApiClient.ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('data')) obj.data = _ApiClient.ApiClient.convertToType(data['data'], [_InstrumentData.InstrumentData]);
    }
    return obj;
  }
}

/**
 * Allowed values for the <code>status</code> property.
 * @enum {String}
 * @readonly
 */
exports.GetOptionContractResponse = GetOptionContractResponse;
GetOptionContractResponse.StatusEnum = {
  /**
   * value: "success"
   * @const
   */
  success: "success",
  /**
   * value: "error"
   * @const
   */
  error: "error"
};
/**
 * @member {module:model/GetOptionContractResponse.StatusEnum} status
 */
GetOptionContractResponse.prototype.status = undefined;

/**
 * Response data for option contracts
 * @member {Array.<module:model/InstrumentData>} data
 */
GetOptionContractResponse.prototype.data = undefined;
