// upstox-js-sdk/dist/model/Margin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Margin = void 0;
var _ApiClient = require("../ApiClient");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The Margin model module.
 * @module model/Margin
 * @version v0
 */
class Margin {
  /**
   * Constructs a new <code>Margin</code>.
   * Response data for instrument margin details
   * @alias module:model/Margin
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>Margin</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Margin} obj Optional instance to populate.
   * @return {module:model/Margin} The populated <code>Margin</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new Margin();
      if (data.hasOwnProperty('span_margin')) obj.spanMargin = _ApiClient.ApiClient.convertToType(data['span_margin'], 'Number');
      if (data.hasOwnProperty('exposure_margin')) obj.exposureMargin = _ApiClient.ApiClient.convertToType(data['exposure_margin'], 'Number');
      if (data.hasOwnProperty('equity_margin')) obj.equityMargin = _ApiClient.ApiClient.convertToType(data['equity_margin'], 'Number');
      if (data.hasOwnProperty('net_buy_premium')) obj.netBuyPremium = _ApiClient.ApiClient.convertToType(data['net_buy_premium'], 'Number');
      if (data.hasOwnProperty('additional_margin')) obj.additionalMargin = _ApiClient.ApiClient.convertToType(data['additional_margin'], 'Number');
      if (data.hasOwnProperty('total_margin')) obj.totalMargin = _ApiClient.ApiClient.convertToType(data['total_margin'], 'Number');
      if (data.hasOwnProperty('tender_margin')) obj.tenderMargin = _ApiClient.ApiClient.convertToType(data['tender_margin'], 'Number');
    }
    return obj;
  }
}

/**
 * Upfront margin mandatory by exchange for derivatives trade applicable on only FNO trade
 * @member {Number} spanMargin
 */
exports.Margin = Margin;
Margin.prototype.spanMargin = undefined;

/**
 * Based on ELM percentage values provided by exchange, applicable on only FNO trade
 * @member {Number} exposureMargin
 */
Margin.prototype.exposureMargin = undefined;

/**
 * Margin applicable for any equity trade
 * @member {Number} equityMargin
 */
Margin.prototype.equityMargin = undefined;

/**
 * Option premium required.
 * @member {Number} netBuyPremium
 */
Margin.prototype.netBuyPremium = undefined;

/**
 * Application margin applicable on MCX FNO trade for certain commodities
 * @member {Number} additionalMargin
 */
Margin.prototype.additionalMargin = undefined;

/**
 * Total margin required for the basket
 * @member {Number} totalMargin
 */
Margin.prototype.totalMargin = undefined;

/**
 * Tender margin
 * @member {Number} tenderMargin
 */
Margin.prototype.tenderMargin = undefined;
