// upstox-js-sdk/dist/model/MarketStatusData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MarketStatusData = void 0;
var _ApiClient = require("../ApiClient");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The MarketStatusData model module.
 * @module model/MarketStatusData
 * @version v0
 */
class MarketStatusData {
  /**
   * Constructs a new <code>MarketStatusData</code>.
   * Response data for market live status
   * @alias module:model/MarketStatusData
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>MarketStatusData</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/MarketStatusData} obj Optional instance to populate.
   * @return {module:model/MarketStatusData} The populated <code>MarketStatusData</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new MarketStatusData();
      if (data.hasOwnProperty('exchange')) obj.exchange = _ApiClient.ApiClient.convertToType(data['exchange'], 'String');
      if (data.hasOwnProperty('status')) obj.status = _ApiClient.ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('last_updated')) obj.lastUpdated = _ApiClient.ApiClient.convertToType(data['last_updated'], 'Number');
    }
    return obj;
  }
}

/**
 * @member {String} exchange
 */
exports.MarketStatusData = MarketStatusData;
MarketStatusData.prototype.exchange = undefined;

/**
 * @member {String} status
 */
MarketStatusData.prototype.status = undefined;

/**
 * @member {Number} lastUpdated
 */
MarketStatusData.prototype.lastUpdated = undefined;
