// upstox-js-sdk/dist/model/MultiOrderError.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MultiOrderError = void 0;
var _ApiClient = require("../ApiClient");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.62
 *
 * Do not edit the class manually.
 *
 */

/**
 * The MultiOrderError model module.
 * @module model/MultiOrderError
 * @version v0
 */
class MultiOrderError {
  /**
   * Constructs a new <code>MultiOrderError</code>.
   * Error details for multi order request
   * @alias module:model/MultiOrderError
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>MultiOrderError</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/MultiOrderError} obj Optional instance to populate.
   * @return {module:model/MultiOrderError} The populated <code>MultiOrderError</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new MultiOrderError();
      if (data.hasOwnProperty('errorCode')) obj.errorCode = _ApiClient.ApiClient.convertToType(data['errorCode'], 'String');
      if (data.hasOwnProperty('message')) obj.message = _ApiClient.ApiClient.convertToType(data['message'], 'String');
      if (data.hasOwnProperty('propertyPath')) obj.propertyPath = _ApiClient.ApiClient.convertToType(data['propertyPath'], 'String');
      if (data.hasOwnProperty('invalidValue')) obj.invalidValue = _ApiClient.ApiClient.convertToType(data['invalidValue'], Object);
      if (data.hasOwnProperty('error_code')) obj.errorCode = _ApiClient.ApiClient.convertToType(data['error_code'], 'String');
      if (data.hasOwnProperty('property_path')) obj.propertyPath = _ApiClient.ApiClient.convertToType(data['property_path'], 'String');
      if (data.hasOwnProperty('invalid_value')) obj.invalidValue = _ApiClient.ApiClient.convertToType(data['invalid_value'], Object);
      if (data.hasOwnProperty('correlation_id')) obj.correlationId = _ApiClient.ApiClient.convertToType(data['correlation_id'], 'String');
    }
    return obj;
  }
}

/**
 * Unique code for the error state
 * @member {String} errorCode
 */
exports.MultiOrderError = MultiOrderError;
MultiOrderError.prototype.errorCode = undefined;

/**
 * Verbose message for the error state
 * @member {String} message
 */
MultiOrderError.prototype.message = undefined;

/**
 * Path to property failing validation
 * @member {String} propertyPath
 */
MultiOrderError.prototype.propertyPath = undefined;

/**
 * Invalid value for the property failing validation
 * @member {Object} invalidValue
 */
MultiOrderError.prototype.invalidValue = undefined;

/**
 * @member {String} errorCode
 */
MultiOrderError.prototype.errorCode = undefined;

/**
 * @member {String} propertyPath
 */
MultiOrderError.prototype.propertyPath = undefined;

/**
 * @member {Object} invalidValue
 */
MultiOrderError.prototype.invalidValue = undefined;

/**
 * A unique identifier for tracking individual orders within the batch
 * @member {String} correlationId
 */
MultiOrderError.prototype.correlationId = undefined;
