// upstox-js-sdk/dist/model/OptionStrikeData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OptionStrikeData = void 0;
var _ApiClient = require("../ApiClient");
var _PutCallOptionChainData = require("./PutCallOptionChainData");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The OptionStrikeData model module.
 * @module model/OptionStrikeData
 * @version v0
 */
class OptionStrikeData {
  /**
   * Constructs a new <code>OptionStrikeData</code>.
   * Response data for option chain data
   * @alias module:model/OptionStrikeData
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>OptionStrikeData</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/OptionStrikeData} obj Optional instance to populate.
   * @return {module:model/OptionStrikeData} The populated <code>OptionStrikeData</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new OptionStrikeData();
      if (data.hasOwnProperty('expiry')) obj.expiry = _ApiClient.ApiClient.convertToType(data['expiry'], 'Date');
      if (data.hasOwnProperty('pcr')) obj.pcr = _ApiClient.ApiClient.convertToType(data['pcr'], 'Number');
      if (data.hasOwnProperty('strike_price')) obj.strikePrice = _ApiClient.ApiClient.convertToType(data['strike_price'], 'Number');
      if (data.hasOwnProperty('underlying_key')) obj.underlyingKey = _ApiClient.ApiClient.convertToType(data['underlying_key'], 'String');
      if (data.hasOwnProperty('underlying_spot_price')) obj.underlyingSpotPrice = _ApiClient.ApiClient.convertToType(data['underlying_spot_price'], 'Number');
      if (data.hasOwnProperty('call_options')) obj.callOptions = _PutCallOptionChainData.PutCallOptionChainData.constructFromObject(data['call_options']);
      if (data.hasOwnProperty('put_options')) obj.putOptions = _PutCallOptionChainData.PutCallOptionChainData.constructFromObject(data['put_options']);
    }
    return obj;
  }
}

/**
 * @member {Date} expiry
 */
exports.OptionStrikeData = OptionStrikeData;
OptionStrikeData.prototype.expiry = undefined;

/**
 * @member {Number} pcr
 */
OptionStrikeData.prototype.pcr = undefined;

/**
 * @member {Number} strikePrice
 */
OptionStrikeData.prototype.strikePrice = undefined;

/**
 * @member {String} underlyingKey
 */
OptionStrikeData.prototype.underlyingKey = undefined;

/**
 * @member {Number} underlyingSpotPrice
 */
OptionStrikeData.prototype.underlyingSpotPrice = undefined;

/**
 * @member {module:model/PutCallOptionChainData} callOptions
 */
OptionStrikeData.prototype.callOptions = undefined;

/**
 * @member {module:model/PutCallOptionChainData} putOptions
 */
OptionStrikeData.prototype.putOptions = undefined;
