// upstox-js-sdk/dist/model/PostMarginResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PostMarginResponse = void 0;
var _ApiClient = require("../ApiClient");
var _MarginData = require("./MarginData");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The PostMarginResponse model module.
 * @module model/PostMarginResponse
 * @version v0
 */
class PostMarginResponse {
  /**
   * Constructs a new <code>PostMarginResponse</code>.
   * @alias module:model/PostMarginResponse
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>PostMarginResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PostMarginResponse} obj Optional instance to populate.
   * @return {module:model/PostMarginResponse} The populated <code>PostMarginResponse</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new PostMarginResponse();
      if (data.hasOwnProperty('status')) obj.status = _ApiClient.ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('data')) obj.data = _MarginData.MarginData.constructFromObject(data['data']);
    }
    return obj;
  }
}

/**
 * Allowed values for the <code>status</code> property.
 * @enum {String}
 * @readonly
 */
exports.PostMarginResponse = PostMarginResponse;
PostMarginResponse.StatusEnum = {
  /**
   * value: "success"
   * @const
   */
  success: "success",
  /**
   * value: "error"
   * @const
   */
  error: "error"
};
/**
 * @member {module:model/PostMarginResponse.StatusEnum} status
 */
PostMarginResponse.prototype.status = undefined;

/**
 * @member {module:model/MarginData} data
 */
PostMarginResponse.prototype.data = undefined;
