// upstox-js-sdk/dist/model/PutCallOptionChainData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PutCallOptionChainData = void 0;
var _ApiClient = require("../ApiClient");
var _AnalyticsData = require("./AnalyticsData");
var _MarketData = require("./MarketData");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The PutCallOptionChainData model module.
 * @module model/PutCallOptionChainData
 * @version v0
 */
class PutCallOptionChainData {
  /**
   * Constructs a new <code>PutCallOptionChainData</code>.
   * @alias module:model/PutCallOptionChainData
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>PutCallOptionChainData</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PutCallOptionChainData} obj Optional instance to populate.
   * @return {module:model/PutCallOptionChainData} The populated <code>PutCallOptionChainData</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new PutCallOptionChainData();
      if (data.hasOwnProperty('instrument_key')) obj.instrumentKey = _ApiClient.ApiClient.convertToType(data['instrument_key'], 'String');
      if (data.hasOwnProperty('market_data')) obj.marketData = _MarketData.MarketData.constructFromObject(data['market_data']);
      if (data.hasOwnProperty('option_greeks')) obj.optionGreeks = _AnalyticsData.AnalyticsData.constructFromObject(data['option_greeks']);
    }
    return obj;
  }
}

/**
 * @member {String} instrumentKey
 */
exports.PutCallOptionChainData = PutCallOptionChainData;
PutCallOptionChainData.prototype.instrumentKey = undefined;

/**
 * @member {module:model/MarketData} marketData
 */
PutCallOptionChainData.prototype.marketData = undefined;

/**
 * @member {module:model/AnalyticsData} optionGreeks
 */
PutCallOptionChainData.prototype.optionGreeks = undefined;
