// upstox-js-sdk/dist/model/TradeHistoryResponseTradeData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TradeHistoryResponseTradeData = void 0;
var _ApiClient = require("../ApiClient");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.54
 *
 * Do not edit the class manually.
 *
 */

/**
 * The TradeHistoryResponseTradeData model module.
 * @module model/TradeHistoryResponseTradeData
 * @version v0
 */
class TradeHistoryResponseTradeData {
  /**
   * Constructs a new <code>TradeHistoryResponseTradeData</code>.
   * @alias module:model/TradeHistoryResponseTradeData
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>TradeHistoryResponseTradeData</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/TradeHistoryResponseTradeData} obj Optional instance to populate.
   * @return {module:model/TradeHistoryResponseTradeData} The populated <code>TradeHistoryResponseTradeData</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new TradeHistoryResponseTradeData();
      if (data.hasOwnProperty('exchange')) obj.exchange = _ApiClient.ApiClient.convertToType(data['exchange'], 'String');
      if (data.hasOwnProperty('segment')) obj.segment = _ApiClient.ApiClient.convertToType(data['segment'], 'String');
      if (data.hasOwnProperty('option_type')) obj.optionType = _ApiClient.ApiClient.convertToType(data['option_type'], 'String');
      if (data.hasOwnProperty('quantity')) obj.quantity = _ApiClient.ApiClient.convertToType(data['quantity'], 'Number');
      if (data.hasOwnProperty('amount')) obj.amount = _ApiClient.ApiClient.convertToType(data['amount'], 'Number');
      if (data.hasOwnProperty('trade_id')) obj.tradeId = _ApiClient.ApiClient.convertToType(data['trade_id'], 'String');
      if (data.hasOwnProperty('trade_date')) obj.tradeDate = _ApiClient.ApiClient.convertToType(data['trade_date'], 'String');
      if (data.hasOwnProperty('transaction_type')) obj.transactionType = _ApiClient.ApiClient.convertToType(data['transaction_type'], 'String');
      if (data.hasOwnProperty('scrip_name')) obj.scripName = _ApiClient.ApiClient.convertToType(data['scrip_name'], 'String');
      if (data.hasOwnProperty('strike_price')) obj.strikePrice = _ApiClient.ApiClient.convertToType(data['strike_price'], 'String');
      if (data.hasOwnProperty('expiry')) obj.expiry = _ApiClient.ApiClient.convertToType(data['expiry'], 'String');
      if (data.hasOwnProperty('price')) obj.price = _ApiClient.ApiClient.convertToType(data['price'], 'Number');
      if (data.hasOwnProperty('isin')) obj.isin = _ApiClient.ApiClient.convertToType(data['isin'], 'String');
      if (data.hasOwnProperty('symbol')) obj.symbol = _ApiClient.ApiClient.convertToType(data['symbol'], 'String');
      if (data.hasOwnProperty('instrument_token')) obj.instrumentToken = _ApiClient.ApiClient.convertToType(data['instrument_token'], 'String');
    }
    return obj;
  }
}

/**
 * @member {String} exchange
 */
exports.TradeHistoryResponseTradeData = TradeHistoryResponseTradeData;
TradeHistoryResponseTradeData.prototype.exchange = undefined;

/**
 * @member {String} segment
 */
TradeHistoryResponseTradeData.prototype.segment = undefined;

/**
 * @member {String} optionType
 */
TradeHistoryResponseTradeData.prototype.optionType = undefined;

/**
 * @member {Number} quantity
 */
TradeHistoryResponseTradeData.prototype.quantity = undefined;

/**
 * @member {Number} amount
 */
TradeHistoryResponseTradeData.prototype.amount = undefined;

/**
 * @member {String} tradeId
 */
TradeHistoryResponseTradeData.prototype.tradeId = undefined;

/**
 * @member {String} tradeDate
 */
TradeHistoryResponseTradeData.prototype.tradeDate = undefined;

/**
 * @member {String} transactionType
 */
TradeHistoryResponseTradeData.prototype.transactionType = undefined;

/**
 * @member {String} scripName
 */
TradeHistoryResponseTradeData.prototype.scripName = undefined;

/**
 * @member {String} strikePrice
 */
TradeHistoryResponseTradeData.prototype.strikePrice = undefined;

/**
 * @member {String} expiry
 */
TradeHistoryResponseTradeData.prototype.expiry = undefined;

/**
 * @member {Number} price
 */
TradeHistoryResponseTradeData.prototype.price = undefined;

/**
 * @member {String} isin
 */
TradeHistoryResponseTradeData.prototype.isin = undefined;

/**
 * @member {String} symbol
 */
TradeHistoryResponseTradeData.prototype.symbol = undefined;

/**
 * @member {String} instrumentToken
 */
TradeHistoryResponseTradeData.prototype.instrumentToken = undefined;
